lib/_tls_wrap.js

```diff
@@ -1098,14 +1098,24 @@ function onConnectEnd() {
   }
 }
 
+let warnOnAllowUnauthorized = true;
+
 // Arguments: [port,] [host,] [options,] [cb]
 exports.connect = function connect(...args) {
   args = normalizeConnectArgs(args);
   var options = args[0];
   var cb = args[1];
+  const allowUnauthorized = '0' === process.env.NODE_TLS_REJECT_UNAUTHORIZED;
+
+  if (allowUnauthorized && warnOnAllowUnauthorized) {
+    warnOnAllowUnauthorized = false;
+    process.emitWarning('Setting the NODE_TLS_REJECT_UNAUTHORIZED ' +
+                        'environment variable to \'0\' is considered ' +
+                        'insecure.');
```

Comment:
«considered insecure» is too broad, imo.

Perhaps something among the lines «makes tls connections and https requests insecure»?

Reply:
Possibly even add "by disabling certificate verification" or so, because my experience is that people using this option almost never actually know what it *does*... and have just enabled it because a code example suggested it / a StackOverflow answer told them to / their boss told them to / etc.

Reply:
Yes, «makes tls connections and https requests insecure by disabling certificate verification» looks much better to me :-).